### PR TITLE
forms-gen-anti-forgery-tokens

### LINKED ISSUE #3688: 2.0 Automatic use of anti-forgery tokens
Add this everywhere we talk about the tokens.

https://github.com/aspnet/Announcements/issues/233

### PR DESCRIPTION
**Pull request by Rick-Anderson**
[Internal Review URL](https://review.docs.microsoft.com/en-us/aspnet/core/security/anti-request-forgery?branch=pr-en-us-3743)
fixes #3688 2.0 Automatic use of anti-forgery tokens #3688

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@fiyazbinhasan  please review

File: aspnetcore/security/anti-request-forgery.md [hunk 1]
```diff
@@ -41,30 +40,67 @@ An example of a CSRF attack:
 Notice that the form action posts to the vulnerable site, not to the malicious site. This is the “cross-site” part of CSRF.
 
 4. The user clicks the submit button. The browser automatically includes the authentication cookie for the requested domain (the vulnerable site in this case) with the request.
+5. The request runs on the server with the user’s authentication context and can do anything that an authenticated user is allowed to do.
 
-5. The request runs on the server with the user’s authentication context, and can do anything that an authenticated user is allowed to do.
+This example requires the user to click the form button. The malicious page could run a script that automatically submits the form or sends a form submission as an AJAX request. The form could  be hidden using CSS so the user never realizes it's present. Using SSL does not prevent a CSRF attack, the malicious site can send an `https://` request.
```
Comment (NTaylorMullen) [inline aspnetcore/security/anti-request-forgery.md]: These seem like bullet points in sentence form.

Comment (NTaylorMullen) [inline aspnetcore/security/anti-request-forgery.md]: `The form could  be` has an extra space. Few places in this doc has the same. I'd do a find and replace for two spaces that aren't after a period.

File: aspnetcore/security/anti-request-forgery.md [hunk 2]
```diff
@@ -41,30 +40,67 @@ An example of a CSRF attack:
...
-* Logging off of web sites when they have finished using them
-* Clearing their browser's cookies periodically
+* Logging off of web sites when they have finished using them.
+* Clearing their browser's cookies periodically.
 
 However, CSRF vulnerabilities are fundamentally a problem with the web app, not the end user.
 
 ## How does ASP.NET Core MVC address CSRF?
 
+In ASP.NET Core MVC 2.0 the [FormTagHelper](xref:mvc/views/working-with-forms#the-form-tag-helper] injects anti-forgery tokens for HTML form elements. For example, the following markup in a Razor file will automatically generate anti-forgery tokens:
```
Comment (NTaylorMullen) [inline aspnetcore/security/anti-request-forgery.md]: The link isn't valid.

File: aspnetcore/security/anti-request-forgery.md [hunk 3]
```diff
@@ -41,30 +40,67 @@ An example of a CSRF attack:
...
+  * The action attribute is empty. ( `action="") OR
+  * The action attribute is not supplied. (<form method="post">)
+
+You can disable automatic generation of anti-forgery tokens for HTML form elements by:
+
+* Explicitly disabling `asp-antiforgery`. For example
+
+ ```html
+  <form method="post" asp-antiforgery="false">
+  /form>
```
Comment (NTaylorMullen) [inline aspnetcore/security/anti-request-forgery.md]: missing `<`

File: aspnetcore/security/anti-request-forgery.md [hunk 4]
```diff
@@ -77,7 +113,7 @@ The most common approach to defending against CSRF attacks is the synchronizer t
 }
 ```
 
-You can also explicitly add an antiforgery token to a ``<form>`` element you create without using tag helpers or HTML helpers by using ``@Html.AntiForgeryToken()``:
+You can explicitly add an antiforgery token to a ``<form>`` element without using tag helpers or HTML helpers with ``@Html.AntiForgeryToken``:
```
Comment (NTaylorMullen) [inline aspnetcore/security/anti-request-forgery.md]: `@Html.AntiForgeryToken` is essentially an `HtmlHelper`. I'd reword this

File: aspnetcore/security/anti-request-forgery.md [hunk 5]
```diff
@@ -291,26 +330,23 @@ The [IAntiForgeryAdditionalDataProvider](https://docs.microsoft.com/aspnet/core/
 
 ## Fundamentals
 
-CSRF attacks rely on the default browser behavior of sending cookies associated with a domain with every request made to that domain. These cookies are stored within the browser. They frequently include session cookies for authenticated users. Cookie-based authentication is the most popular form of authentication used by web applications. However, token-based authentication systems have been growing in popularity in recent years, especially for SPAs and other "smart client" scenarios.
+CSRF attacks rely on the default browser behavior of sending cookies associated with a domain with every request made to that domain. These cookies are stored within the browser. They frequently include session cookies for authenticated users. Cookie-based authentication is a popular form of authentication. token-based authentication systems have been growing in popularity, especially for SPAs and other "smart client" scenarios.
```
Comment (NTaylorMullen) [inline aspnetcore/security/anti-request-forgery.md]: `token-based` => `Token-based`

**Comment by Rick-Anderson** [comment]
@NTaylorMullen  addressed all the issues.

**Comment by blowdart** [comment]
Nothing changed that concerns me.

**Comment by fiyazbinhasan** [comment]
@Rick-Anderson  reviewed. looks good 👍

**Pull request merged**